analyzers/rspec/vbnet/S4583_vb.net.html

```diff
@@ -0,0 +1,81 @@
+called. This is why you should always pair <code>BeginInvoke</code> with an <code>EndInvoke</code> to complete your asynchronous call.</p>
+<p>This rule raises an issue when:</p>
+<ul>
+  <li> the <code>BeginInvoke</code> method is called without any callback and it is not paired with a call to <code>EndInvoke</code> in the same
+  block. </li>
+  <li> a callback with a single parameter of type <code>IAsyncResult</code> doesn't contain a call to <code>EndInvoke</code>. </li>
+</ul>
+<h2>Noncompliant Code Example</h2>
+<p>BeginInvoke without callback</p>
+<pre>
+Public Delegate Function AsyncMethodCaller() As String
+
+Public Class Sample
+
+    Public Sub DoSomething()
+        Dim Example As New AsyncExample()
+        Dim Caller As New AsyncMethodCaller(Example.SomeMethod)
+        ' Initiate the asynchronous call.
+        Dim Result As IAsyncResult = Caller.BeginInvoke(Nothing, Nothing) ' Noncompliant - Not paired With EndInvoke
+    End Sub
+
+End Class
+</pre>
+<p>BeginInvoke with callback</p>
+<pre>
+Public Delegate Function AsyncMethodCaller() As String
+
+Public Class Sample
+
+    Public Sub DoSomething()
+        Dim Example As New AsyncExample()
+        Dim Caller As New AsyncMethodCaller(Example.SomeMethod)
+        ' Initiate the asynchronous call.
+        Dim Result As IAsyncResult = Caller.BeginInvoke(New AsyncCallback(Sub(ar)
+                                                                          End Sub), Nothing) ' Noncompliant - Not paired With EndInvoke
+    End Sub
+
+End Class
+</pre>
+<h2>Compliant Solution</h2>
+<p>BeginInvoke without callback</p>
+<pre>
+Public Delegate Function AsyncMethodCaller() As String
+
+Public Class Sample
+
+    Public Function DoSomething() As String
+        Dim Example As New AsyncExample()
+        Dim Caller As New AsyncMethodCaller(Example.SomeMethod)
+        ' Initiate the asynchronous call.
+        Dim Result As IAsyncResult = Caller.BeginInvoke(Nothing, Nothing)
+        ' ...
+        Return Caller.EndInvoke(Result)
+    End Function
+
+End Class
+</pre>
+<p>BeginInvoke with callback</p>
+<pre>
+Public Delegate Function AsyncMethodCaller() As String
+
+Public Class Sample
+
+    Public Sub DoSomething()
+        Dim Example As New AsyncExample()
+        Dim Caller As New AsyncMethodCaller(Example.SomeMethod)
+        ' Initiate the asynchronous call.
+        Dim Result As IAsyncResult = Caller.BeginInvoke(New AsyncCallback(Sub(ar)
+                                                                              ' Call EndInvoke to retrieve the results.
+                                                                              Dim Ret As String = Caller.EndInvoke(ar)
+                                                                              ' ...
+                                                                          End Sub), Nothing) ' Noncompliant - Not paired With EndInvoke
```

Comment:
Why is this here in the Compliant section?

Reply:
Copy/Paste issue